reflex/components/datadisplay/table.py

```diff
@@ -77,8 +78,20 @@ def create(cls, *children, headers=None, **props) -> Component:
 
         Returns:
             The table header component.
+
+        Raises:
+            TypeError: If headers are not of type list or type tuple.
         """
         if len(children) == 0:
+            if (
+                (
+                    isinstance(headers, Var)
+                    and types.get_base_class(headers.type_) not in (list, tuple)
+                )
+                or not isinstance(headers, Var)
+                and types.get_base_class(type(headers)) not in (list, tuple)
+            ):
+                raise TypeError("table headers should be a list or tuple")
```

Comment:
i think it would be nice to extract this logic out into a helper function for readability

Reply:
got it